enricher/api/src/main/java/io/fabric8/maven/enricher/api/AbstractHealthCheckEnricher.java

```diff
@@ -0,0 +1,90 @@
+package io.fabric8.maven.enricher.api;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+
+/**
+ * Enriches containers with health check probes.
+ */
+public abstract class AbstractHealthCheckEnricher extends BaseEnricher {
+
+    public AbstractHealthCheckEnricher(EnricherContext buildContext, String name) {
+        super(buildContext, name);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            if (item instanceof Deployment) {
+                Deployment deployment = (Deployment) item;
+                Container container = getContainer(deployment);
+                if (container != null) {
+                    if (container.getReadinessProbe() == null) {
+                        Probe probe = getReadinessProbe();
+                        if (probe != null) {
+                            log.info("Adding readiness " + describe(probe));
+                            container.setReadinessProbe(probe);
+                        }
+                    }
+
+                    if (container.getLivenessProbe() == null) {
+                        Probe probe = getLivenessProbe();
+                        if (probe != null) {
+                            log.info("Adding liveness " + describe(probe));
+                            container.setLivenessProbe(probe);
+                        }
+                    }
+                }
+            }
+        }
+        builder.withItems(items);
+    }
+
+    private Container getContainer(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers().size() > 0) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            return containers.get(containers.size() - 1); // last one
```

Comment:
I've also a doubt about this part. I did it so because the original "enricher" updated only the last image.
But I'm not aware of the cases where the f-m-p can build more than 1 container and put all them in a deployment, and why only the last one should be enriched. Thoughts?